dspace-api/src/main/java/org/dspace/app/util/GoogleBitstreamComparator.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.util;
+
+import org.apache.log4j.Logger;
+import org.dspace.content.Bitstream;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by frederic on 24/04/17.
+ */
+public class GoogleBitstreamComparator implements Comparator<Bitstream>{
+
+    private final static Logger log = Logger.getLogger(GoogleBitstreamComparator.class);
+
+    HashMap<String, Integer> priorityMap = new HashMap<>();
+
+    public GoogleBitstreamComparator(Map<String, String> googleScholarSettings){
+        String[] types;
+        if (googleScholarSettings.containsKey("citation.prioritized_types")){
+            types = splitAndTrim(googleScholarSettings.get("citation.prioritized_types"));
+        } else {
+            log.warn("Please define citation.prioritized_types in google-metadata.properties");
+            types = new String[0];
+        }
+        int priority = 1;
+        for(String s: types){
+            String[] mimetypes;
+            if (googleScholarSettings.containsKey("citation.mimetypes." + s)){
```

Comment:
Rather than using both a `prioritized_types` and mapped `mimetypes` configurations, I'd recommend simply using the type names defined in the DSpace Bitstream Format registry (see the "short name" column): https://wiki.duraspace.org/display/DSDOC5x/Metadata+and+Bitstream+Format+Registries#MetadataandBitstreamFormatRegistries-DefaultBitstreamFormatRegistry

This would mean this could use a single configuration, e.g. `google.citation.prioritized_types` to reference the prioritized bitstream formats. It'd also be more in line with how the MediaFilters are configured for various file formats: https://github.com/DSpace/DSpace/blob/master/dspace/config/dspace.cfg#L377